etna/models/nn/tft_native/tft.py

```diff
@@ -0,0 +1,691 @@
+                "time_varying_reals_encoder": dict(),
+                "time_varying_reals_decoder": dict(),
+            }
+            total_length = len(values_target)
+            total_sample_length = encoder_length + decoder_length
+
+            if total_sample_length + start_idx > total_length:
+                return None
+
+            sample["decoder_target"] = values_target[encoder_length:].reshape(-1, 1)  # (decoder_length, 1)
+
+            for feature in self.static_reals:
+                sample["static_reals"][feature] = df[[feature]].values[:1]  # (1, 1)
+
+            for feature in self.static_categoricals:
+                sample["static_categoricals"][feature] = df[[feature]].values[:1]  # (1, 1)
+
+            for feature in self.time_varying_categoricals_encoder:
+                sample["time_varying_categoricals_encoder"][feature] = df[[feature]].values[
+                    1:encoder_length
+                ]  # (encoder_length-1, 1)
+
+            for feature in self.time_varying_categoricals_decoder:
+                sample["time_varying_categoricals_decoder"][feature] = df[[feature]].values[
+                    encoder_length:
+                ]  # (decoder_length, 1)
+
+            for feature in self.time_varying_reals_encoder:
+                sample["time_varying_reals_encoder"][feature] = df[[feature]].values[
+                    1:encoder_length
+                ]  # (encoder_length-1, 1)
+
+            for feature in self.time_varying_reals_decoder:
+                sample["time_varying_reals_decoder"][feature] = df[[feature]].values[
+                    encoder_length:
+                ]  # (decoder_length, 1)
+
+            return sample
+
+        start_idx = 0
+        while True:
+            batch = _make(
+                values_target=values_target,
+                df=df,
+                start_idx=start_idx,
+                encoder_length=encoder_length,
+                decoder_length=decoder_length,
+            )
+            if batch is None:
+                break
+            yield batch
+            start_idx += 1
+
+    def configure_optimizers(self) -> "torch.optim.Optimizer":
+        """Optimizer configuration."""
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.lr, **self.optimizer_params)
+        return optimizer
+
+
+class TFTNativeModel(DeepBaseModel):
+    """TFT model.
+
+    Note
+    ----
+    This model requires ``torch`` extension to be installed.
+    Read more about this at :ref:`installation page <installation>`.
+    """
+
+    def __init__(
+        self,
+        encoder_length: int,
+        decoder_length: int,
+        n_heads: int = 2,
+        num_layers: int = 2,
+        dropout: float = 0.0,
+        hidden_size: int = 16,
+        lr: float = 1e-3,
+        static_categoricals: List[str] = [],
+        static_reals: List[str] = [],
+        time_varying_categoricals_encoder: List[str] = [],
```

Comment:
Why do we have separate lists for encoder and decoder? It is for the case when feature is known/unknown in the future?

Reply:
Can't we define them automatically based on metadata of TSDataset? We know which columns are regressors + which are categoricals